submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,354 @@
+}
+
+type EgressIPSpec struct {
+    // User can explicitly request a specific GlobalIP and this should be a valid IPaddress
+    // from globalnetCIDR allocated to the Cluster. If the requested globalIP falls outside
+    // of the globalnetCIDR or is already allocated (either fully or partially), then Status
+    // field will be set to Error with appropriate message.
+    // If this field is not specified, a single globalIP will be allocated.
+    // This field cannot be changed through updates.
+    // +optional
+    GlobalIP      string  `json:"globalIP,omitempty"`
+
+    // The number of globalIP's requested. Globalnet Controller will allocate the requested
+    // number of contiguous GlobalIPs for this GlobalnetEgressIP object.
+    // User can specify this field while omitting GlobalIP. In such cases, Globalnet will
+    // auto-allocate the requested number of contiguous GlobalIPs.
+    // If unspecified, NumGlobalIPs defaults to 1.
+    // +optional
+    NumGlobalIPs  int     'json:"numGlobalIPs",omitempty"`
+
+    // Selects the Namespaces to which this GlobalnetEgressIP object applies.
+    // If an empty namespaceSelector: {} is configured, it selects all namespaces.
+    // If an empty namespaceSelector: {} is configured along with empty PodSelector, it applies
+    // to whole Cluster.
+    // On the other hand, if you omit specifying namespaceSelector, it does not select any
+    // namespaces and selects only Pods from the namespace where the GlobalnetEgressIP is
+    // deployed to.
+    // +optional
+    NamespaceSelector metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+    // Selects the pods whose label match the definition. This is an optional field and
+    // in case it is not set, it results in all the Pods selected from the NamespaceSelector.
+    // In case it is set, its intersected with NamespaceSelector, and only Pods that match the
+    // PodSelector and present in the NamespaceSelector will have the GlobalIPs as EgressIPs.
+    // Either namespaceSelector or podSelector have to be specified. If both are omitted,
+    // its considered as an error.
+    // When a Pod or Namespace matches multiple GlobalnetEgressIP objects, there is no guarantee
+    // which of the globalIP address that are assigned to GlobalnetEgressIP will be used.
+    // +optional
+    PodSelector metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+
+const (
+    // GlobalnetEgressIPSuccess means that Globalnet was able to successfully
+    // allocate the GlobalIP as requested in GlobalnetEgressIP object.
+    GlobalnetEgressIPSuccess GlobalnetEgressIPStatus = "Success"
+
+    // GlobalnetEgressIPError means that Globalnet was unable to allocate the
+    // GlobalIP for the requested GlobalnetEgressIP object.
+    GlobalnetEgressIPError GlobalnetEgressIPStatus = "Error"
+)
+
+type EgressIPStatus struct {
+    // Status is one of {"Success", "Error"}
+    Status GlobalnetEgressIPStatus `json:"status,omitempty"`
```

Comment:
I think we should move this to a list of conditions, like it's standard in k8s. So the admin can see the progress of the object. 

May be there will be not much progress in this type of object though, what do you think?

Reply:
I'm not able to think of any progress for this type except for `Success` or `Error`. Hence did not use the `Conditions`

Reply:
Yes probably the Success/Error could be a condition since it fits the bill, other fields not so much.